Fix/actor details screen

baraa0abd opened the pull request:
fixed bugs : 
132- The spacing between genres and the date is too large.
133- The "Read More" text color is incorrect.
134- The padding around the actor image is incorrect.
135- The spacing between the "Known For" label and the Lazy Row is too large.
136- The spacing between the star and the rate is too large.
137- Padding is missing in the "Known For" list.
138- The rate format is incorrect.
139- The star icon used for the rate is incorrect.

baraa0abd committed:
Refactor: Remove empty search layout from ForYouAndExploreScreen

This commit removes the explicit empty search layout handling within `ForYouAndExploreScreen.kt`. The logic to display an image when `forYouMovies` is empty has been removed. The parent composable or a higher-level component is now expected to handle the empty state.

baraa0abd committed:
Refactor: Update MovioVerticalCard and ForYouAndExploreScreen

This commit introduces changes to the `MovioVerticalCard` and `ForYouAndExploreScreen` composables.

**MovioVerticalCard:**
- Removed the `paddingValue` parameter as it's no longer used.
- Ensured the `description` text handles long strings by truncating with an ellipsis and limiting to a single line.

**ForYouAndExploreScreen:**
- Removed the top padding from the `Column` composables that display loading and error states for a more consistent layout.
- Added bottom padding to the `LazyRow` displaying "For You" movies.
- Removed explicit height setting from the "For You" movies `LazyRow`.

baraa0abd committed:
Fix: Prevent long description overflow in MovioVerticalCard preview

The description text in the `VerticalCardPreview` has been shortened to prevent it from overflowing and causing visual issues in the preview.

baraa0abd committed:
Fix: Padding in ForYouAndExploreScreen

This commit adds padding to the `LazyRow` in `ForYouAndExploreScreen.kt`.

Specifically, it adds a 12dp padding around the `LazyRow` within the `ForYouScreen` composable.

baraa0abd committed:
Merge branch 'refs/heads/develop' into fix/18-movie-title-max-lines-incorrect-

baraa0abd committed:
Refactor: Update RateIcon and remove unused files

- Updated `RateIcon` to format the rate to one decimal place by truncating, not rounding.
- Removed unused commented-out code from `MovioEpisodesCard.kt`.
- Deleted `movieRow.kt` as it's no longer used.

baraa0abd committed:
Merge branch 'Issue/15-unexpected-screen-when-loading-search' into fix/18-movie-title-max-lines-incorrect-

# Conflicts:
#	presentation/src/main/java/com/madrid/presentation/screens/searchScreen/features/recentSearchLayout/ForYouAndExploreScreen.kt

baraa0abd committed:
Refactor: Remove empty state for "For You" movies

This commit removes the empty state handling from the "For You" movies section in `ForYouAndExploreScreen.kt`. The UI no longer displays an image when `forYouMovies` is empty.

baraa0abd committed:
Refactor: Format movie ratings to one decimal place

This commit introduces a utility function `toOneDecimalPlace` for `Double` values.
This function is used in `SimilarMoviesScreen.kt` to format the movie rating displayed in the `MovieCard`, ensuring it's shown with a single decimal place.
A try-catch block is included in the new function to handle potential exceptions during formatting, falling back to the original string representation if an error occurs.

baraa0abd committed:
Refactor: Enhance TopCastSection UI and text handling

This commit introduces several UI improvements to the `TopCastSection` composable:

- **Padding and Spacing:**
    - Added horizontal padding of `16.dp` to the main row containing the title and "See all" button.
    - Increased vertical spacing between the title row and the cast list to `16.dp`.
    - Added horizontal content padding of `16.dp` to the `LazyRow` for cast members.
    - Increased vertical spacing between the cast member image and name to `8.dp`.

- **Cast Member Item:**
    - Set a fixed width of `80.dp` for `CastMemberItem` to ensure consistent spacing between items.
    - Updated the text style for the cast member name to `Theme.textStyle.label.smallRegular12` for a better fit.
    - Enabled `maxLines = 2` and `overflow = TextOverflow.Ellipsis` for cast member names to handle long names gracefully.
    - Set `textAlign = TextAlign.Center` for cast member names.
    - Applied `modifier = Modifier.fillMaxWidth()` to the cast member name `MovioText` to allow text to wrap within the fixed width.

- **Preview:**
    - Updated the `TopCastSectionPreview` to include cast members with longer names and different name lengths to better test the text handling changes.

baraa0abd committed:
Fix: Adjust bottom padding in ForYouAndExploreScreen

The bottom padding for the LazyRow in `ForYouAndExploreScreen.kt` has been reduced from `32.dp` to `10.dp`.

baraa0abd committed:
Fix: Adjust bottom padding in ForYouAndExploreScreen

The bottom padding for the LazyRow in `ForYouAndExploreScreen.kt` has been reduced from `32.dp` to `10.dp`.

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into fix/18-movie-title-max-lines-incorrect-

# Conflicts:
#	presentation/src/main/java/com/madrid/presentation/screens/searchScreen/features/recentSearchLayout/ForYouAndExploreScreen.kt

baraa0abd committed:
Refactor: Update TopCastSection UI

Increased the size of the cast member image in `TopCastSection` from `60.dp` to `86.dp x 87.dp`.

baraa0abd committed:
Refactor: Remove Bottom Sheet components and update dependencies

This commit removes the following Bottom Sheet components from the `presentation` module:
- `MovieListsBottomSheet.kt`
- `RatingMovieBottomSheet.kt`
- `SaveMovieBottomSheet.kt`
- `ShareMovieBottomSheet.kt`

**Dependency Updates:**
- Removed `androidx.material3` and `androidx.material3.android` dependencies from `presentation/build.gradle.kts` as they are no longer directly used after the bottom sheet removals.

**Component Relocation:**
- Moved `ExpandableDescription.kt` from `presentation/screens/detailsScreen/componant` to `designSystem/component`.

**UI Updates:**
- Replaced `Text` with `MovioText` in `ForYouAndExploreScreen.kt` and `TrendingLayout.kt` for consistency.
- Updated `MovioButton.kt` in `designSystem` to:
    - Add an optional `backgroundColor` parameter. If `null`, the button will have no background and minimal padding, suitable for text buttons.
    - Introduce a new `MovioTextButton` composable for creating text-only buttons.

**Impact:**
- The removed Bottom Sheet components were specific to the `detailsScreen` and are no longer part of the application.
- The `ExpandableDescription` component is now part of the `designSystem` for broader reusability.
- UI components in search and home screens now use the consistent `MovioText` component.
- `MovioButton` offers more flexibility in styling, particularly for text buttons.

baraa0abd committed:
Refactor: Format rate display

This commit introduces a `RateFormatter` utility to consistently format media ratings across the application.

**Changes:**

- Created `RateFormatter.kt` with functions to format `Double` and `String` rates.
    - The formatter truncates the rate to one decimal place.
    - If the rate is a whole number after truncation, it's displayed as an integer.
- Integrated `RateFormatter` in:
    - `DetailsMovieViewModel`
    - `SeriesDetailsViewModel`
    - `UiStateMapper` (for movie and series mapping)
- Removed redundant rate formatting logic from `RateIcon.kt` and `SimilarMoviesScreen.kt` as it's now handled by `RateFormatter`.
- `SimilarMovie` data class in `SimilarMoviesScreen.kt` now expects a `String` for `rating`.
- Removed an unused log statement from `SeriesDetailsScreen.kt`.

baraa0abd committed:
Merge branch 'develop' into fix/18-movie-title-max-lines-incorrect-

baraa0abd committed:
Refactor: Format rate display

This commit introduces a `RateFormatter` utility to consistently format media ratings across the application.

**Changes:**

- Created `RateFormatter.kt` with functions to format `Double` and `String` rates.
    - The formatter truncates the rate to one decimal place.
    - If the rate is a whole number after truncation, it's displayed as an integer.
- Integrated `RateFormatter` in:
    - `DetailsMovieViewModel`
    - `SeriesDetailsViewModel`
    - `UiStateMapper` (for movie and series mapping)
- Removed redundant rate formatting logic from `RateIcon.kt` and `SimilarMoviesScreen.kt` as it's now handled by `RateFormatter`.
- `SimilarMovie` data class in `SimilarMoviesScreen.kt` now expects a `String` for `rating`.
- Removed an unused log statement from `SeriesDetailsScreen.kt`.

baraa0abd committed:
127- The top cast actor's name is too big and pushes the elements and make the spacing between too big
128- in top actor screen The spacing between the items is too large.
129- The background color is incorrect.
130- The spacing between the image and the text is too small

baraa0abd committed:
131- The top bar color is incorrect.(fixed)

baraa0abd committed:
133- The "Read More" text is misaligned and its color is incorrect.

baraa0abd committed:
136- The spacing between the star and the rate is too large.

baraa0abd committed:
Merge branch 'develop' into fix/actor-details-screen

# Conflicts:
#	designSystem/src/main/java/com/madrid/designSystem/component/TextWithReadMore.kt
#	presentation/src/main/java/com/madrid/presentation/component/TopCastSection.kt
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/castDetails/ActorDetails.kt
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/detailsMovieScreen/MovieDetailScreen.kt
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/seriesDetails/SeriesDetailsScreen.kt
#	presentation/src/main/java/com/madrid/presentation/screens/detailsScreen/similarMedia/SimilarMoviesScreen.kt
#	presentation/src/main/java/com/madrid/presentation/viewModel/detailsViewModel/ActorDetailsViewModel.kt
#	presentation/src/main/java/com/madrid/presentation/viewModel/detailsViewModel/DetailsMovieViewModel.kt
#	presentation/src/main/java/com/madrid/presentation/viewModel/detailsViewModel/SeriesDetailsViewModel.kt

baraa0abd committed:
Refactor: Enhance UI and Code Readability in Details Screens

This commit introduces several improvements to the UI and code structure of the details screens.

**Series Details Screen:**
- Utilized `SeriesDetailsHeader` for a more consistent header presentation.
- Replaced `MovioText` with `CustomTextTitel` for section titles.
- Added an icon for the runtime information.

**Actor Details Screen:**
- Improved the layout by using `LazyVerticalGrid` for a more structured display of actor's movies.
- Added a "Known For" section title.
- Used `GridItemSpan` to make the actor details header span across all columns.

**TextWithReadMore Component:**
- Refined the "Read More" functionality by using `buildAnnotatedString` and `withStyle` for better text styling and click handling.
- Leveraged string resources for "Read More" and "Read Less" text.

**SeriesDetailsViewModel:**
- Integrated `RateFormatter` and `formatDuration` for consistent data presentation.

Alaakhaled101 commented on presentation/src/main/java/com/madrid/presentation/component/movioCards/RateIcon.kt:
```diff
@@ -29,17 +28,19 @@ fun RateIcon(
 ) {
     Box(
         modifier = modifier
+            .padding(top = 4.dp, end= 4.dp)
```
remove padding and add it in card component

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into fix/actor-details-screen

baraa0abd committed:
Refactor based on Review : Adjust padding in MovioVerticalCard and RateIcon

- Removed `paddingValue` from `MovioVerticalCard` calls in `AllMediaLayout`, `MoviesLayout`, and `TvShowsLayout` as it's now handled internally.
- Added `paddingValue` parameter to `MovioVerticalCard` with a default of `0.dp`. This padding is applied to the top and end of the `Row` containing the rate icon.
- Applied `paddingValue = 8.dp` to `MovioVerticalCard` within `CustomHorizontalCard`.
- Removed explicit top and end padding from the `RateIcon`'s `Box` modifier, as this is now handled by the parent `MovioVerticalCard`.
- Changed `TopCastDetailsScreen`'s `LazyVerticalGrid` to use `GridCells.Adaptive(minSize = 101.dp)` instead of `GridCells.Fixed(3)`.

baraa0abd committed:
Merge branch 'develop' into fix/actor-details-screen

# Conflicts:
#	presentation/src/main/java/com/madrid/presentation/viewModel/detailsViewModel/SeriesDetailsViewModel.kt

baraa0abd committed:
Refactor: Update padding and grid layout in UI components

This commit refactors padding in various UI components and modifies the grid layout in `TopCastDetailsScreen`.

**Padding Changes:**
- Added `paddingValue = 8.dp` to `MovioVerticalCard` in `TvShowsLayout.kt`, `MoviesLayout.kt`, and `AllMediaLayout.kt`.
- Removed `paddingValue` parameter from `MovioVerticalCard` composable function and its usage within `CustomHorizontalCard.kt`.
- Applied padding directly within the `RateIcon` composable (`.padding(top = 4.dp, end= 4.dp)`).

**Grid Layout Change:**
- Changed `LazyVerticalGrid` in `TopCastDetailsScreen.kt` to use `GridCells.Fixed(3)` instead of `GridCells.Adaptive(minSize = 101.dp)`.

baraa0abd committed:
Refactor: Remove unused paddingValue from MediaCard

The `paddingValue` parameter in the `MediaCard` composable was not being used and has been removed. The padding is now handled by the `modifier` parameter.

This change affects the following layouts:
- `AllMediaLayout.kt`
- `MoviesLayout.kt`
- `TvShowsLayout.kt`
